### PR TITLE
Migrate builds to GHA

### LINKED ISSUE #68: Test with GitHub Actions
This repo would be a good candidate to test using ~Azure Pipelines~ GitHub Actions. We have 24 jobs and ~AP gives 10~ GHA gives 20 parallel builds. Travis CI only gives 5.

The main `python-pillow/Pillow` repo's ~AP~ GHA config can be used as a basis, it's running the Docker tests.

* ~https://github.com/python-pillow/Pillow/blob/master/azure-pipelines.yml~
* ~https://github.com/python-pillow/Pillow/blob/master/.azure-pipelines/jobs/test-docker.yml~
* https://github.com/python-pillow/Pillow/blob/master/.github/workflows/test-docker.yml

~AP~ GHA now has cron too:

~https://docs.microsoft.com/en-us/azure/devops/pipelines/build/triggers?view=azure-devops&tabs=yaml~

https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onschedule

Once done, we can either remove or cut down the `.travis.yml` to a minimal, fast set (to finish quicker than ~AP~ GHA); as a backup and because it's an easy and more familiar thing for contributors to enable.

### PR DESCRIPTION
**Pull request by nulano**
Fixes #68 

For https://github.com/python-pillow/Pillow/issues/5028

The script is almost an exact copy from Travis, except updated to GHA syntax; there is just one small change due to file permissions: `sudo chmod a+w .` added at the start of the build and the `sudo chown` command moved later.

Docker login details need to be stored as a Github Secret in `DOCKER_USERNAME` and `DOCKER_PASSWORD`.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+
+name: Docker Images
```
Comment (hugovk) [inline .github/workflows/build.yml:2]:
```suggestion
name: Docker images
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      - name: Prepare build
+        run: |
+          sudo apt-get update && sudo apt-get install -qyy debootstrap
+          if [ "${{ matrix.target }}" = "latest" ]; then
+            git submodule update --remote Pillow
+            (cd Pillow && git checkout master)
+          fi
+          sudo chown -R 1000 $(pwd)
+
+      - name: Build Image
```
Comment (hugovk) [inline .github/workflows/build.yml:42]:
```suggestion
      - name: Build image
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          cd "${{ matrix.image }}" && sudo chmod a+w . && make update
+          if [[ -n "$DOCKER_USERNAME" ]]; then
+            echo $DOCKER_PASSWORD | docker login -u $DOCKER_USERNAME --password-stdin
+          fi
+          make build BRANCH=master && make test BRANCH=master
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Push Image
```
Comment (hugovk) [inline .github/workflows/build.yml:53]:
```suggestion
      - name: Push image
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        run: make push-${{ matrix.image }} BRANCH=master
+
+      - name: Post build
+        if: always()
+        run: sudo chown -R $(id -u) $(pwd)
+
+  success:
+    needs: build
+    runs-on: ubuntu-latest
+    name: Build Successful
```
Comment (hugovk) [inline .github/workflows/build.yml:64]:
```suggestion
    name: Build successful
```

**Review by hugovk — APPROVED** [review]
(Consistency nits)

Thanks!

**Comment by hugovk** [comment]
For reference, example builds:

* Before: https://travis-ci.com/github/python-pillow/docker-images/builds/198202805
* After: https://github.com/nulano/docker-images/actions/runs/346077582

---

@wiredfool Please could you review and add the credentials as secrets?

* https://github.com/python-pillow/docker-images/settings/secrets

File: .github/mergify.yml [hunk 5]
```diff
@@ -3,7 +3,7 @@ pull_request_rules:
     conditions:
       - "#approved-reviews-by>=1"
       - label=automerge
-      - status-success=Travis CI - Pull Request
+      - status-success=Build Successful
```
Comment (nulano) [inline .github/mergify.yml:6]:
```suggestion
      - status-success=Build successful
```

**Comment by nulano** [comment]
As a test/demonstration, I have also pushed the built images to my Docker account and tested here: https://github.com/nulano/Pillow/actions/runs/346068049/workflow#L44

**Comment by hugovk** [comment]
Let's merge and add credentials later. Thanks!

**Comment by wiredfool** [comment]
Sorry, this disappeared into my backlog, I’ll try to dog out the creds.

**Pull request merged**